test/k8sT/Policies.go

```diff
@@ -1115,6 +1117,191 @@ var _ = Describe("K8sPolicyTest", func() {
 			_ = kubectl.Exec(cmd)
 		})
 
+		SkipContextIf(helpers.DoesNotExistNodeWithoutCilium, "validates ingress CIDR-dependent L4 (FromCIDR + ToPorts)", func() {
+			var (
+				outsideNodeName, outsideIP string // k8s3 node (doesn't have agent running)
+
+				backendPod   v1.Pod // The pod that k8s3 node is hitting
+				backendPodIP net.IP
+
+				hostNodeName       string // Node that backendPod ends up on
+				hostIPOfBackendPod net.IP
+
+				policyVerdictAllowRegex, policyVerdictDenyRegex *regexp.Regexp
+			)
+
+			BeforeAll(func() {
+				RedeployCiliumWithMerge(kubectl, ciliumFilename, daemonCfg,
+					map[string]string{
+						"global.tunnel":                 "disabled",
+						"global.autoDirectNodeRoutes":   "true",
+						"global.endpointRoutes.enabled": "true",
+
+						// Needed because of
+						// https://github.com/cilium/cilium/issues/12141
+						"global.masquerade":   "false",
+						"global.devices":      "",
+						"global.hostFirewall": "disabled",
```

Comment:
Per-endpoint routes are usually used when doing direct routing so I just did that without much thought. I've removed it as it is not explicitly required.

As for #12141, it turns out I can enable masquerading (I think I misinterpreted the aforementioned issue). However, I need to explicity do the following:

```
"global.devices":      "",
"global.hostFirewall": "false",
```

for the tests to pass.

I don't understand why clearing out the `devices` is required, but if I don't clear them, the tests fail by the wrong identity to incoming traffic (sets it to `WORLD`). I think this is what Jarno was seeing in this thread: https://cilium.slack.com/archives/CDKG8NNHK/p1592425114171600.

As for the host firewall, do you think it should be disabled, or that it shouldn't matter? I can see why it's having an impact, but I am not sure if that's ideal.

Reply:
> I don't understand why clearing out the `devices` is required, but if I don't clear them, the tests fail by the wrong identity to incoming traffic (sets it to `WORLD`). I think this is what Jarno was seeing in this thread: https://cilium.slack.com/archives/CDKG8NNHK/p1592425114171600.

Yep, I think we have a bug in the logic we use to derive the source security ID in bpf_host. That identity is then passed to the bpf_lxc programs (and used in the host firewall). I haven't had time to take a second look yet.

> As for the host firewall, do you think it should be disabled, or that it shouldn't matter? I can see why it's having an impact, but I am not sure if that's ideal.

The host firewall shouldn't matter here. However, it requires `devices` to be set, so we'll have to disable it as long as `devices` aren't set.

> As for #12141, it turns out I can enable masquerading (I think I misinterpreted the aforementioned issue). However, I need to explicity do the following:
> 
> ```
> "global.devices":      "",
> "global.hostFirewall": "false",
> ```
> 
> for the tests to pass.

The other way around should work too: disabling `masquerade` and enabling `hostFirewall`+`devices`. It might be best to keep it that way (i.e., host firewall disabled) as I'll need to add another test anyway to test FromCIDR+ToPorts as a host policy; I will likely test the reverse option in that other test (i.e., masquerading disabled).